File: DSA/LeetCode/Arrays/3075-Maximize-Happiness-of-Selected-Children.js
/**
 * @param {number[]} happiness
 * @param {number} k
 * @return {number}
 */
var maximumHappinessSum = function (happiness, k) {


    happiness.sort((a, b) => b - a)
    let decreaseby = 0;
    let sum = 0
    for (let i = 0; i < k; i++) {
        let child = happiness[i]
        sum += Math.max(0, child - decreaseby)
        decreaseby++
    }



    return sum

};

File: DSA/LeetCode/Arrays/3075-Maximize-Happiness-of-Selected-Children.py
class Solution(object):
    def maximumHappinessSum(self, happiness, k):
        """
        :type happiness: List[int]
        :type k: int
        :rtype: int
        """
        


        happiness.sort(reverse=True)
        decreaseby = 0
        sum = 0
        for i in range(k):
            child = happiness[i]
            sum +=  max(0, child - decreaseby)
            decreaseby += 1
        



        return sum

